src/flow/darcy_flow_mh.cc

```diff
@@ -654,316 +644,166 @@ void  DarcyMH::get_parallel_solution_vector(Vec &vec)
-
-            // update matrix for weights in BDDCML
-            if ( typeid(*ls) == typeid(LinSys_BDDC) ) {
-               int ind = tmp_rows[1];
-               // there is -value on diagonal in block C!
-               double new_val = local_vb[0];
-               static_cast<LinSys_BDDC*>(ls)->diagonal_weights_set_value( ind, new_val );
-            }
-
             if (n_schur_compls == 2) {
                 // for 2. Schur: N dim edge is conected with N dim element =>
                 // there are nz between N dim edge and N-1 dim edges of the element
-
-                ls->mat_set_values(nsides, edge_rows, 1, tmp_rows+1, zeros);
-                ls->mat_set_values(1, tmp_rows+1, nsides, edge_rows, zeros);
+                
+                // be carefull with edge-edge entry
+                ls->mat_set_values(nsides, edge_rows, 1, &neigh_edge_row, zeros);
+                ls->mat_set_values(1, &neigh_edge_row, nsides, edge_rows, zeros);
 
                 // save all global edge indices to higher positions
                 tmp_rows[2+i] = tmp_rows[1];
             }
         }
 
-
         // add virtual values for schur complement allocation
         uint n_neigh;
         switch (n_schur_compls) {
         case 2:
             n_neigh = ele_ac.full_iter()->n_neighs_vb;
             // Connections between edges of N+1 dim. elements neighboring with actual N dim element 'ele'
-        	OLD_ASSERT(n_neigh*n_neigh<1000, "Too many values in E block.");
+            OLD_ASSERT(n_neigh*n_neigh<1000, "Too many values in E block.");
             ls->mat_set_values(ele_ac.full_iter()->n_neighs_vb, tmp_rows+2,
                     ele_ac.full_iter()->n_neighs_vb, tmp_rows+2, zeros);
 
-        case 1: // included also for case 2
-            // -(C')*(A-)*B block and its transpose conect edge with its elements
-            ls->mat_set_values(1, &ele_row, ele_ac.n_sides(), edge_rows, zeros);
-            ls->mat_set_values(ele_ac.n_sides(), edge_rows, 1, &ele_row, zeros);
-            // -(C')*(A-)*C block conect all edges of every element
-            ls->mat_set_values(ele_ac.n_sides(), edge_rows, ele_ac.n_sides(), edge_rows, zeros);
+            // Here I cannot add values to the same positions, since it uses ADD_VALUE, when counting nnz.
+//         case 1: // included also for case 2
+//             // -(C')*(A-)*B block and its transpose conect edge with its elements
+//             ls->mat_set_values(1, &ele_row, ele_ac.n_sides(), edge_rows, zeros);
+//             ls->mat_set_values(ele_ac.n_sides(), edge_rows, 1, &ele_row, zeros);
+//             // -(C')*(A-)*C block conect all edges of every element
+//             ls->mat_set_values(ele_ac.n_sides(), edge_rows, ele_ac.n_sides(), edge_rows, zeros);
         }
-    }    
-    
-    if (balance_ != nullptr && fill_matrix)
-        balance_->finish_flux_assembly(water_balance_idx_);
-
-
+    }
 
+    // alloc edge diagonal entries
```

Comment:
I still don't like some parts of the allocation code. I allocate the whole local matrix for [velocity,press] x [velocity,press] but I cannot do it so easily for edges, since they are shared by elements.